### ISSUE TITLE
Blocking=False causes error.

### ISSUE THREAD

**Issue by beregond** [bluele/redis-semaphore#7]
When passing `blocking=False` to semaphore - instead of acquiring the lock, I get `IndexError`:

```
In [1]: from redis_semaphore import Semaphore

In [2]: from onrecruit.concurrency import redis_connection

In [3]: s1 = Semaphore(redis_connection, count=1, namespace='aasdfasdfasdf', blocking=False)

In [4]: s1.acquire()
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-4-477f1a8eb196> in <module>()
----> 1 s1.acquire()

/usr/local/lib/python3.5/site-packages/redis_semaphore/__init__.py in acquire(self, timeout, target)
     59                 raise NotAvailable
     60 
---> 61         token = pair[1]
     62         self._local_tokens.append(token)
     63         self.client.hset(self.grabbed_key, token, self.current_time)

IndexError: index out of range

In [5]: s1.acquire()
---------------------------------------------------------------------------
NotAvailable                              Traceback (most recent call last)
<ipython-input-5-477f1a8eb196> in <module>()
----> 1 s1.acquire()

/usr/local/lib/python3.5/site-packages/redis_semaphore/__init__.py in acquire(self, timeout, target)
     57             pair = self.client.lpop(self.available_key)
     58             if pair is None:
---> 59                 raise NotAvailable
     60 
     61         token = pair[1]
```

From this point on lock is always acquired (it always raises `NotAvailable`) and you cannot `release()` it. Only `reset()` helps.

Tested on python 3.5, with redis 2.8 and 3.0 (same behaviour), Redis-Semaphore==0.2.0 and redis==2.10.5

**Comment by beregond** [comment]
Fixed in #8

The cause was changing `blocking` on the fly, so case when `lpop` actually returned anything other than `None` was never checked.

**Comment by bluele** [comment]
Merged #8
